api/agent/agent.go

```diff
@@ -696,24 +696,28 @@ func (s *hotSlot) exec(ctx context.Context, call *call) error {
 	}
 }
 
-func (s *hotSlot) dispatch(ctx context.Context, call *call) chan error {
-	ctx, span := trace.StartSpan(ctx, "agent_dispatch_httpstream")
-	defer span.End()
-
-	// TODO we can't trust that resp.Write doesn't timeout, even if the http
-	// client should respect the request context (right?) so we still need this (right?)
-	errApp := make(chan error, 1)
+var removeHeaders = map[string]bool{
+	"connection":        true,
+	"keep-alive":        true,
+	"trailer":           true,
+	"transfer-encoding": true,
+	"te":                true,
+	"upgrade":           true,
+	"authorization":     true,
+}
```

Comment:
thanks. there's a decent list https://golang.org/src/net/http/httputil/reverseproxy.go?s=#L145 as well (couple missing, but that's ok, just putting in for reference...)